app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -158,12 +182,14 @@ public boolean onOptionsItemSelected(MenuItem item)
 
         return super.onOptionsItemSelected(item);
     }
-
+    
     @Override
     protected void onPostResume()
     {
         super.onPostResume();
+        if (map != null) map.clear();
         getMapPreferences();
+        getCurrentAssignment();
```

Comment:
It is on the main thread in other places, but it probably would be good to get if off of it. I'll see about moving it.